lib/internal/modules/cjs/loader.js

```diff
@@ -302,20 +305,20 @@ function readPackage(requestPath) {
 }
 
 function readPackageScope(checkPath) {
-  const rootSeparatorIndex = checkPath.indexOf(path.sep);
+  const rootSeparatorIndex = StringPrototypeIndexOf(checkPath, sep);
   let separatorIndex;
-  while (
-    (separatorIndex = checkPath.lastIndexOf(path.sep)) > rootSeparatorIndex
-  ) {
-    checkPath = checkPath.slice(0, separatorIndex);
-    if (checkPath.endsWith(path.sep + 'node_modules'))
+  do {
+    separatorIndex = StringPrototypeLastIndexOf(checkPath,
+                                                sep, checkPath.length - 2);
```

Comment:
Hmm, I don't immediately understand why this should be the case (`checkPath.length - 2`).

Reply:
Because the last character is a separator, we start the search from the second last character.

Reply:
I've switched to adding the last separator separately.